### PR TITLE
SPECIES NUKING 2023: Makes tongues handle liked/disliked food instead of species datum

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

Alright, let's get real, handling food tastes with the species datum is hacky at best - This is a continuation of my nuking of the species datum, which currently handles way too much of human mobs' behaviors. 

While moving that to the tongue isn't exactly a perfect solution, it allows for more interesting behavior than the species datum, especially now that I added getter procs to get foodtypes liked/disliked by the mob and such.

## Why It's Good For The Game

Makes tongue transplants more appealing and emergent, since they fully control tastes and not just taste sensitivity which is pretty much cosmetic.

## Changelog

:cl:
refactor: Liking/disliking food is now handled by the tongue organ, not the species. Also, having a failing tongue means you can't taste food properly!
add: You can read peoples' tongues now, if you have the entrail reading skillchip.
/:cl:

### CODE REVIEW

**Comment by ChungusGamer666** [comment]
three fucking species datum gutting prs at once lets fucking go
species nuke 2023

**Comment by MTandi** [comment]
Uh, not sure. They say that there is a gene responsible for loving bitter coffee and dark chocolate.

**Comment by ChungusGamer666** [comment]
> Uh, not sure. They say that there is a gene responsible for loving bitter coffee and dark chocolate.

I know that this isn't really accurate in any way to real life, but being able to swap tongues and get the person's tastes for food is just a really cool, emergent gameplay thing.
I'll consider keeping some functionality in the species datum if I have to, but gameplay wise there is just a lot of potential to deferring most - if not all - tastes to the tongue.

**Comment by Kapu1178** [comment]
Personally this makes more sense in the Stomach. Since it also defines what is toxic to them.

**Comment by ChungusGamer666** [comment]
> Personally this makes more sense in the Stomach. Since it also defines what is toxic to them.

Thought about moving toxic foodtypes to the stomach, but then I realized that toxic food just behaves as well, food you hate.
Will keep it as a tongue thing.

**Comment by ChungusGamer666** [comment]
image
it be working

**Comment by ChungusGamer666** [comment]
it's OVER

**Comment by ChungusGamer666** [comment]
unit test failures that aint my fault, love that shit

**Comment by ChungusGamer666** [comment]
bullshit!!!

**Comment by ChungusGamer666** [comment]
can a maintainer rerun tests ok thx

**Comment by ZephyrTFA** [comment]
stop breaking shit

**Comment by YakumoChen** [comment]
Does that mean if you remove your tongue you can eat basically anything? 

I think the stomach makes a lot more sense, taste and appetite are far more linked to the guts in real life, the tongue just gets signals based on those preferences.

**Comment by MisterBomb7** [comment]
How does this work with the voicebox tongue?

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 1]
```diff
@@ -13,13 +12,10 @@
...
 	changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_PRIDE | MIRROR_MAGIC | RACE_SWAP | ERT_SPAWN | SLIME_EXTRACT
 	species_language_holder = /datum/language_holder/felinid
-	disliked_food = GROSS | CLOTH | RAW
-	liked_food = SEAFOOD | ORANGES | BUGS | GORE
-	var/original_felinid = TRUE //set to false for felinids created by mass-purrbation
 	payday_modifier = 0.75
 	ass_image = 'icons/ass/asscat.png'
 	family_heirlooms = list(/obj/item/toy/cattoy)
-	examine_limb_id = SPECIES_HUMAN
+	var/original_felinid = TRUE //set to false for felinids created by mass-purrbation
```
Comment (Time-Green) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm:18]: Autodoc :)

File: code/modules/reagents/reagent_containers/cups/_cup.dm [hunk 2]
```diff
@@ -25,34 +25,33 @@
...
-			H.adjust_disgust(11 + 15 * fraction)
-			H.add_mood_event("gross_food", /datum/mood_event/gross_food)
-		else if(drink_type & H.dna.species.liked_food)
-			to_chat(H,span_notice("I love this taste!"))
-			H.adjust_disgust(-5 + -2.5 * fraction)
-			H.add_mood_event("fav_food", /datum/mood_event/favorite_food)
-
+// all this copypaste from the edible component is quite deplorable, why the fuck does this not use the edible component?
+/obj/item/reagent_containers/cup/proc/checkLiked(fraction, mob/eater)
+	if(last_check_time + 50 > world.time)
```
Comment (Time-Green) [inline code/modules/reagents/reagent_containers/cups/_cup.dm:30]:
```suggestion
	if(last_check_time + 5 SECONDS > world.time)
```

File: code/modules/reagents/reagent_containers/cups/_cup.dm [hunk 3]
```diff
@@ -25,34 +25,33 @@
...
-		else if(drink_type & H.dna.species.disliked_food)
-			to_chat(H,span_notice("That didn't taste very good..."))
-			H.adjust_disgust(11 + 15 * fraction)
-			H.add_mood_event("gross_food", /datum/mood_event/gross_food)
-		else if(drink_type & H.dna.species.liked_food)
-			to_chat(H,span_notice("I love this taste!"))
-			H.adjust_disgust(-5 + -2.5 * fraction)
-			H.add_mood_event("fav_food", /datum/mood_event/favorite_food)
-
+// all this copypaste from the edible component is quite deplorable, why the fuck does this not use the edible component?
```
Comment (Time-Green) [inline code/modules/reagents/reagent_containers/cups/_cup.dm:28]: Please just replace with a normal autodoc comment (you can still recommend putting it on edible component)

**Review by Time-Green — CHANGES_REQUESTED** [review]
Looks good, minor nitpicks

**Comment by ChungusGamer666** [comment]
> Does that mean if you remove your tongue you can eat basically anything?

No. Toxic foodtypes without a tongue default to TOXIC.

> I think the stomach makes a lot more sense, taste and appetite are far more linked to the guts in real life (the gut flora, to be exact), the tongue just gets signals based on those preferences.

I don't agree. Ultimately, your tongue is what receives taste, and this is how I implemented it. The assumption is that every species evolved a tongue that likes certain types of food because it suits their stomach regardless.

> Also removing your stomach means you can't just bypass food restrictions.

If you go through the effort of removing your tongue just to be able to eat your disliked foodtypes... I am fine with it. Also this is the behavior with tongues before this PR, as tongues already apply ageusia when removed. So the precedent is set already.

File: code/datums/quirks/neutral_quirks.dm [hunk 4]
```diff
@@ -110,20 +102,16 @@
...
 
 /datum/quirk/pineapple_liker/remove()
-	var/mob/living/carbon/human/human_holder = quirk_holder
-	var/datum/species/species = human_holder.dna.species
-	species.liked_food &= ~PINEAPPLE
-	UnregisterSignal(human_holder, COMSIG_SPECIES_GAIN)
+	var/obj/item/organ/internal/tongue/tongue = quirk_holder.get_organ_slot(ORGAN_SLOT_TONGUE)
+	if(!tongue)
+		return
+	tongue.liked_foodtypes = initial(tongue.liked_foodtypes)
```
[the remark this answers is not in the document]
Comment (ChungusGamer666) [reply code/datums/quirks/neutral_quirks.dm:114]: absolutely, this gave me a bit of a headache

**Comment by YakumoChen** [comment]
> I don't agree. Ultimately, your tongue is what receives taste, and this is how I implemented it. The assumption is that every species evolved a tongue that likes certain types of food because it suits their stomach regardless.

I don't agree with this. The tongue handles speech, which it's always done, and the stomach handles food. We shouldn't be having one organ do both. Changing your tongue shouldn't give you the burden of another species's foodtypes.

**Comment by ChungusGamer666** [comment]
> 
> I don't agree with this. The tongue handles speech, which it's always done, and the stomach handles food. We shouldn't be having one organ do both. Changing your tongue shouldn't give you the burden of another species's foodtypes.

At this point, it's up to maintainers. But I think the precedent has been set very strongly by the existing tongue ageusia mechanic.

**Comment by ChungusGamer666** [comment]
> How does this work with the voicebox tongue?

Default human tastes.

**Comment by Time-Green** [comment]
You can just fix the merge conflicts on one of the species PR and then ping me on it, to avoid headaches

**Comment by ZephyrTFA** [comment]
I'm fine with liked foods being dictated by what your tongue is even if it may not actually match what is actually toxic to you.
It makes sense and is also realistic.

**Comment by Time-Green** [comment]
oh yeah probably shouldve specified but im fine with it too

**Pull request merged**